modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/introspection/validation/MetadataComponentModelValidatorTestCase.java

```diff
@@ -310,4 +319,23 @@ private ParameterModel getMockKeyPartParam(Object defaultValue, int order) {
     when(param.getModelProperty(MetadataKeyPartModelProperty.class)).thenReturn(of(new MetadataKeyPartModelProperty(order)));
     return param;
   }
+
+  public static class MockResolver implements TypeKeysResolver, OutputTypeResolver {
+
+    @Override
+    public MetadataType getOutputType(MetadataContext context, Object key)
+        throws MetadataResolvingException, ConnectionException {
+      return null;
```

Comment:
Yeah, we could fail at runtime.
Currently we just default to the static type. I thought about throwing an exception, but left it that way just in case we needed a workaround of `dynamic resolver but if key is invalid the default to the static java type that is unknown to me because I'm resolving metadata for multiple operations`